osquery/sql/tests/sql.cpp

```diff
@@ -462,4 +462,256 @@ TEST_F(SQLTests, test_concat_ws_fail) {
   ASSERT_TRUE(!status.ok());
 }
 
+/*
+ * collate version
+ */
+TEST_F(SQLTests, test_collate_version_eq) {
+  QueryData d;
+  auto status = query(
+      "select '1.0' = '1.0' collate version as t0, \
+              '1:0.0' = '1.0.0' collate version as t1, \
```

Comment:
I _assume_ this is true, because it's using `version` and not `version_dpkg` but it ends up seeming a little confusing. Which is, somewhat, a reflection on the underlying system -- We have several flavors of version comparison, and it's hard to organize the tests for them

Reply:
Yeah, I wanted to showcase that without using the `delimiterPrecedence`, different delims will be equal.

Reply:
Maybe a comment about what the test is showing?